### PR TITLE
Update sarif.xsl to support relative paths for the uri of violations.

### PR DESCRIPTION
**Pull request by tren-parasoft**

### CODE REVIEW

File: src/main/resources/xsl/sarif.xsl [hunk 1]
```diff
@@ -573,32 +573,37 @@
     </xsl:template>
     
     <xsl:template name="artifact_location">
-        <xsl:text>"artifactLocation": { "uri": "</xsl:text>
+        <xsl:text>"artifactLocation": {</xsl:text>
```
Comment (yshi-parasoft) [inline src/main/resources/xsl/sarif.xsl:576]: This change broke the functionality at line 612

**Review by yshi-parasoft — DISMISSED** [review]

File: src/main/resources/xsl/sarif.xsl [hunk 2]
```diff
@@ -573,87 +573,124 @@
...
                         <xsl:choose>
                             <xsl:when test="$locNode/@scPath">
-                                <xsl:value-of select="$locNode/@scPath" /><xsl:text>"</xsl:text>
-                                <xsl:variable name="uriBaseId" select="$locNode/@repRef"/>
-                                <xsl:if test="$uriBaseId != ''">
-                                    <xsl:text>, "uriBaseId": "ROOT_</xsl:text><xsl:value-of select="$uriBaseId" /><xsl:text>"</xsl:text>
-                                </xsl:if>
+                                <!-- When source control information is included in report -->
+                                <xsl:variable name="scUri" select="$locNode/@scPath" />
+                                <xsl:variable name="scUriBaseId" select="$locNode/@repRef"/>
```
Comment (rwang-parasoft) [inline src/main/resources/xsl/sarif.xsl:620]: We lose the `ROOT_` prefix.

File: src/main/resources/xsl/sarif.xsl [hunk 3]
```diff
@@ -573,87 +573,124 @@
...
+        <xsl:param name="uri"/>
+        <xsl:param name="uriBaseId"/>
+        <xsl:text>"uri": "</xsl:text>
+        <xsl:value-of select="$uri"/>
+        <xsl:text>", "uriBaseId": "</xsl:text>
+        <xsl:value-of select="$uriBaseId"/>
+        <xsl:text>"</xsl:text>
+    </xsl:template>
+
+    <xsl:template name="getMatchedProjectRootPath_by_loc">
```
Comment (rwang-parasoft) [inline src/main/resources/xsl/sarif.xsl:693]: Should we keep this template? Since it is unused in this file.

Comment (tren-parasoft) [reply src/main/resources/xsl/sarif.xsl:693]: You're right , it's not used anymore, I will remove it.

**Review by rwang-parasoft — APPROVED** [review]

**Pull request merged**